### PR TITLE
Feature gate

### LINKED ISSUE #725: Feature Gating
#### What would you like to be added:

Feature gating creates a list of configurations isolated from the main app, accessible via settings. They can be used to enable/disable app features.

With feature gating, we can release new features for testing directly into the main app and have it disabled. Advanced users can choose to enable features while they are in beta.

#### Why is this needed:

This allows us to move faster bring testing features early to a handful of users. Release early, test early, and iterative early.

### PR DESCRIPTION
**Pull request by fullstackninja864**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
In this PR we are allowing user to opt for  Beta features using  Feature gating of configurations isolated from the main app, accessible via settings > About > Try Beta features . They can be used to enable/disable app features.
With feature gating, we can release new features for testing directly into the main app and have it disabled. Advanced users can choose to enable features while they are in beta.

To use the Feature Gate we have two ways available
1. Using `FeatureGate` HOC
```
<FeatureGate feature='<featureId>'>
  <ChildComponent />
</FeatureGate>
```
2. using `isFeatureAvailable` method
```
const isAvailable = isFeatureAvailable('<featureId>')
```
also when user enable the feature, then in order to show users warning related to risk involved using beta feature we can use `isBetaFeature` method to get boolean value.

#### Which issue(s) does this PR fixes?:

Fixes #725

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [x] Unit tests*
- [ ] Added e2e tests*
- [x] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Settings/screens/FeatureFlagScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  value: boolean
+}
+
+export function FeatureFlagScreen (): JSX.Element {
+  const { featureFlags, enabledFeatures, updateEnabledFeatures } = useFeatureFlagContext()
+  const features = featureFlags.filter((item) => item.stage === 'beta')
+  const [betaFeatures, setBetaFeatures] = useState<BetaFeaturesI []>([])
+
+  const getBetaFeature = (flags: FEATURE_FLAG_ID[]): BetaFeaturesI[] => {
+    return features.map((item: FeatureFlag) => {
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Settings/screens/FeatureFlagScreen.tsx:22]: we are looping here twice (loop on another loop). Maybe we can change it to a better option

**Review by thedoublejay — CHANGES_REQUESTED** [review]
1. Try Beta Feature button has an underline
2. It's too close to components above and below it, maybe add some margin

File: website/src/pages/api/v0/settings/flags.ts [hunk 2]
```diff
@@ -11,8 +11,9 @@ export default async function handle (req: NextApiRequest, res: NextApiResponse<
   await runMiddleware(req, res, cors)
   res.json([{
     id: 'loan',
-    name: 'Decentralized Loans',
-    stage: 'alpha',
-    version: '>=0.12.0'
+    name: 'Loans',
+    stage: 'beta',
```
Comment (thedoublejay) [inline website/src/pages/api/v0/settings/flags.ts:15]: @fullstackninja864 this should be Alpha, don't change it to Beta yet.

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: website/src/pages/api/v0/settings/flags.ts (hunk 2 above)
Comment (fullstackninja864) [reply website/src/pages/api/v0/settings/flags.ts:15]: Sure, will update it

File: mobile-app/app/screens/AppNavigator/screens/Settings/screens/AboutScreen.tsx [hunk 3]
```diff
@@ -103,6 +105,10 @@ const SOCIAL_LINKS: AboutScreenSocialLinks[] = [
 ]
 
 export function AboutScreen (): JSX.Element {
+  const navigation = useNavigation<NavigationProp<SettingsParamList>>()
+  const { featureFlags } = useFeatureFlagContext()
+  const features = featureFlags.filter((item) => item.stage === 'beta')
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Settings/screens/AboutScreen.tsx:110]: `featureFlags.some()` could be a better option here if you're only interested if an item exist

**Review by kyleleow — CHANGES_REQUESTED** [review]
Can i request to add some e2e test cases? We can stub the `getFeatureFlag()` API to simulate different scenario

File: mobile-app/app/contexts/FeatureFlagContext.tsx [hunk 4]
```diff
@@ -20,51 +25,76 @@ export function useFeatureFlagContext (): FeatureFlagContextI {
...
-  const [isAuctionDisplayed, setIsAuctionDisplayed] = useState<boolean>(false)
+  const [enabledFeatures, setEnabledFeatures] = useState<FEATURE_FLAG_ID[]>([])
+
   if (!isError) {
     prefetchPage({})
   }
 
+  function isBetaFeature (featureId: FEATURE_FLAG_ID): boolean {
+    return featureFlags.some((flag) => {
+      return flag.id === featureId && flag.stage === 'beta'
```
Comment (kyleleow) [inline mobile-app/app/contexts/FeatureFlagContext.tsx:44]: need to check for app version if its mobile

File: mobile-app/app/screens/AppNavigator/screens/Settings/screens/AboutScreen.tsx (hunk 3 above)
Comment (fullstackninja864) [reply mobile-app/app/screens/AppNavigator/screens/Settings/screens/AboutScreen.tsx:110]: Will update this, Thanks for suggestions

**Review by kyleleow — APPROVED** [review]

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**